### PR TITLE
make them private

### LINKED ISSUE #111: public vs private  `tensor_to_matrix(tensor_input)`
do we need this to be public?

`def tensor_to_matrix(tensor_input):`
`def matrix_to_tensor`

### PR DESCRIPTION
**Pull request by fnrizzi**

### CODE REVIEW

**Comment by jtencer** [comment]
I'm fine with this change, but I think what it accomplishes is actually to make these methods public but in a different module.  Rather than `romtools.trial_space.tensor_to_matrix` we now have `romtools.tiral_space.utils.tensor_to_matrix`.  If we want to indicate that it's a private function, we should prepend an underscore to the name and place the function in the module that uses it.

**Comment by fnrizzi** [comment]
I thought that the distinction private public for us is just what is documented vs what is not. 
These functions are omitted from being documented in pdoc.  

I am happy to prepend underscore but that still does not make private :)

**Comment by jtencer** [comment]
"python private" I guess.  There's not real privacy in python, but prepending an underscore prevents them from being imported with `from romtools.trial_space.utils import *` and is generally the convention for "keep out".  I haven't used pdoc, but at least for sphinx and underscore has the same effect of omitting it from the documentation.  Does moving them to utils only omit them from the documentation because we don't have any documentation for utils?

**Comment by jtencer** [comment]
Did some googling and it looks like pdoc ignores methods that start with `_`

I don't see anything in [the docs](https://pdoc3.github.io/pdoc/doc/pdoc/#gsc.tab=0) which would make utils explicitly private.

**Comment by jtencer** [comment]
To clarify, I think it's OK to have these functions in utils and to have them be public but undocumented.  I also think it's OK to make them private in the sense of adding an underscore.  However, I do not think that we should have them in utils _and_ have them be private because having them in utils requires that they be imported and used outside of the scope in which they're defined.  If we make them private, we should make them private to the module in which they're used.

Does that make sense?

**Comment by eparish1** [comment]
I agree about prepending with the _. I think if we make these changes we are good.

**Review by jtencer — APPROVED** [review]

**Comment by fnrizzi** [comment]
@jtencer @eparish1 free functions with prepended __ do not work. That works only for class methods. 
So i opted to use a trailing `_impl`. I hope this is fine

**Comment by fnrizzi** [comment]
so @jtencer do you want me to make them private inside the file where they are used or leave them where they are ?

File: tests/romtools/test_trial_space.py [hunk 1]
```diff
@@ -5,6 +5,11 @@
 import romtools.trial_space.utils as utils
 
 
+def local_tensor_to_matrix(tensor_input):
```
Comment (jtencer) [inline tests/romtools/test_trial_space.py]: I think we should just call this `_tensor_to_matrix`

The underscores says that it's just for this module unless the "local" is referring to parallel/serial or something.

Comment (fnrizzi) [reply tests/romtools/test_trial_space.py]: It is because this is a test. I can use _ here too

Comment (jtencer) [reply tests/romtools/test_trial_space.py]: I don't understand.  What does `local` communicate?

Comment (fnrizzi) [reply tests/romtools/test_trial_space.py]: local to this test file , i will change it to `tensor_to_matrix`

Comment (eparish1) [reply tests/romtools/test_trial_space.py]: @fnrizzi I don't see an _impl? Am I understanding it correct that you stuck with the _ and then added the local functions to the test?

Comment (fnrizzi) [reply tests/romtools/test_trial_space.py]: yes i used a single _ for the functions inside the library, and then just use a copy of the function in this test file

**Comment by jtencer** [comment]
I'm good with it now

**Comment by eparish1** [comment]
@fnrizzi please ping me if you'd like me to resolve the conflict.

**Comment by fnrizzi** [comment]
superseeded by #122

**Pull request closed without merging**